### PR TITLE
Add option -limit to condor_history

### PR DESCRIPTION
**Pull request by mschnepf**
condor_history scans per default the complete history. The option ` -limit`  stops `condor_history` after it founds the given number.
This reduces the time to check the status of old jobs.

### CODE REVIEW

**Comment by riga** [comment]
Hi @mschnepf ,
thanks for the PR, appreciate it!
This is actually very useful as status querying can get really slow on some systems. I hope you don't mind that I added another commit to add the `-limit` argument only for htcondor > [v8.3.3](https://research.cs.wisc.edu/htcondor/manual/v8.3.3/condor_history.html). If you're fine with that change, I can merge.

File: law/contrib/htcondor/job.py [hunk 1]
```diff
@@ -143,8 +144,8 @@ def query(self, job_id, pool=None, scheduler=None, user=None, silent=False):
         if scheduler:
             cmd += ["-name", scheduler]
         cmd += ["-long"]
-        # since v8.5.6, one can define the attributes to fetch
-        if self.htcondor_version and self.htcondor_version >= (8, 5, 6):
+        # since v8.5.6 one can define the attributes to fetch
+        if self.htcondor_v856:
```
Comment (riga) [inline law/contrib/htcondor/job.py]: I also hijacked your PR to use the already existing version flag ;)

**Comment by riga** [comment]
I just saw that `condor_q` has the same option. Since we know the maximum number of jobs to expect in the response of `condor_q` (`len(job_ids)`), I don't see why we shouldn't add `-limit` here as well. Objections?

File: law/contrib/htcondor/job.py (hunk 1 above)
Comment (mschnepf) [reply law/contrib/htcondor/job.py]: Sound good :+1:

**Comment by mschnepf** [comment]
Good idea to check which version of HTCondor supports this feature. It worked always for me. :-)
I will add the ` -limit` option also to HTCondor ` condor_q`  command.

**Comment by riga** [comment]
@mschnepf Looks like you committed a different local change.

**Pull request merged**